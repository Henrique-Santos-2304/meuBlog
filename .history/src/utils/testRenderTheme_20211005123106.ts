import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
