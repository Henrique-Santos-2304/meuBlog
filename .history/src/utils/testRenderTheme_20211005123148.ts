import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactElement): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
