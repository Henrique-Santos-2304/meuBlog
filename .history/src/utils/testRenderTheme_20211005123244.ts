import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
