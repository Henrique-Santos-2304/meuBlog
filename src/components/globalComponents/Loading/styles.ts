import styled, { css } from "styled-components";
export const Wrapper = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.secondary};
    width: 100%;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;

    padding: 2rem;

    h1 {
      color: ${theme.colors.primary};
      font-size: ${theme.font.sizes.xxbiglarge};
    }
  `}
`;
