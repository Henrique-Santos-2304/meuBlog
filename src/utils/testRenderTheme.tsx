import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
