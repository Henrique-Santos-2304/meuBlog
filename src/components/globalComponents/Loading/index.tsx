import * as S from "./styles";

const Loading = () => (
  <S.Wrapper>
    <h1>Carregando Dados aguarde sรณ um momento Por favor...</h1>
  </S.Wrapper>
);

export default Loading;
