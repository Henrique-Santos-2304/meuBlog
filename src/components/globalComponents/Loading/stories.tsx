import { Story, Meta } from "@storybook/react";
import Loading from ".";

export default {
  title: "Loading",
  component: Loading,
} as Meta;

export const Basic: Story = (args) => <Loading {...args} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = (args) => <Loading {...args} />;
