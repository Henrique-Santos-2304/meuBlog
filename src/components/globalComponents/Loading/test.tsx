import Loading from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";

describe("<Loading />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<Loading />);
    expect(container).toMatchSnapshot();
  });
  it("should render ", () => {
    renderTheme(<Loading />);
    const text = screen.getByText(
      /Carregando Dados aguarde sรณ um momento Por favor.../i
    );
    expect(text).toBeInTheDocument();
  });
});
