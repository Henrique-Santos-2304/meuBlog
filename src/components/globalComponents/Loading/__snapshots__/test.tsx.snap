// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render  1`] = `
.c0 {
  background: #1e1e1e;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 2rem;
}

.c0 h1 {
  color: #f8f8f8;
  font-size: 3.8rem;
}

<div>
  <div
    class="c0"
  >
    <h1>
      Carregando Dados aguarde sรณ um momento Por favor...
    </h1>
  </div>
</div>
`;
